collect_app/src/main/java/org/odk/collect/android/activities/AppListActivity.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2017 Nafundi
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.odk.collect.android.activities;
+
+import android.app.ListActivity;
+import android.support.annotation.NonNull;
+import android.widget.Button;
+import android.widget.ListView;
+
+import org.odk.collect.android.R;
+import org.odk.collect.android.application.Collect;
+import org.odk.collect.android.database.ActivityLogger;
+
+abstract class AppListActivity extends ListActivity {
+    protected final ActivityLogger logger = Collect.getInstance().getActivityLogger();
+
+    protected boolean areCheckedItems() {
+        return getCheckedCount() > 0;
+    }
+
+    /** The positions and IDs of the checked items */
+    class CheckedItemInfo {
+        final int[] positions;
+        final long[] ids;
+
+        CheckedItemInfo(int[] positions, long[] ids) {
+            this.positions = positions;
+            this.ids = ids;
+        }
+    }
+
+    /** Returns the positions and IDs of the checked items */
+    protected CheckedItemInfo getCheckedItemInfo() {
+        return getCheckedItemInfo(getListView());
+    }
+
+    /** Returns the positions and IDs of the checked items */
+    protected CheckedItemInfo getCheckedItemInfo(ListView lv) {
+        int itemCount = lv.getCount();
+        int checkedItemCount = lv.getCheckedItemCount();
+        int[] checkedPositions = new int[checkedItemCount];
+        long[] checkedIds = new long[checkedItemCount];
+        int resultIndex = 0;
+        for (int posIdx = 0; posIdx < itemCount; ++posIdx) {
```

Comment:
I don't believe using a prefix increment here is any different than using a postfix one. I think postfix is more familiar and that this could be surprising.